### DISCUSSION TITLE
Dev: Deleting pyqt objects?

### DISCUSSION THREAD

**Discussion by real-yfprojects** [borgbase/vorta#1188] [Ideas]
Why do you delete *pyqt* objects? As far as I know garbage collections will do the job. @m3nu  

I am especially interested in this case:
https://github.com/borgbase/vorta/blob/f4a85e82c40d1799ee479fe9b0a2998ebac8b4b4/src/vorta/application.py#L112-L117

**Comment by real-yfprojects** [comment]
By the way, is it intended that the method above is never called at all?

**Reply by m3nu** [reply]
It's called here via a signal: `self.aboutToQuit.connect(self.quit_app_action)`

The cleanup and deletion of objects is needed to avoid a segfault when quitting the app. Python does NOT properly clean up all Qt objects by itself. At least on in v5.
